### PR TITLE
Call Dart plugin registrant if available

### PR DESCRIPTION
**Pull request by blasten**
The engine side of https://github.com/flutter/flutter/issues/52267

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [x] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: runtime/dart_isolate.cc [hunk 1]
```diff
@@ -689,9 +690,14 @@ bool DartIsolate::MarkIsolateRunnable() {
     return false;
   }
 
+  if (Dart_IsError(plugin_registrant_function)) {
```
Comment (jason-simmons) [inline runtime/dart_isolate.cc:693]: Move this down to where `plugin_registrant_function` is assigned in RunFromLibrary

Comment (blasten) [reply runtime/dart_isolate.cc]: Done.

**Review by jason-simmons — APPROVED** [review]

**Review by chinmaygarde — APPROVED** [review]
The test looks correct but I am not sure why the native resolver is failing. You should be able to set a breakpoint at the point of failure though.

**Comment by blasten** [comment]
I had to introduce a new flag in`Setting`, so the new behavior can be turn on from the test. Otherwise, other runtime tests fail due to the undefined native call `PassMessageForDartRegistrantTest`.

**Comment by blasten** [comment]
@chinmaygarde @jason-simmons mentioned that he would prefer I don't add a new flag to settings.h, is there any other way to achieve this behavior? 

The issue is that `_registerPlugins` is called prior to the entrypoint, so unless there's way in the test to run code in the VM prior to the entrypoint, I'm unsure how to get this to work.

**Pull request merged**